src/System.Security.Cryptography.Pkcs/src/Microsoft/Win32/SafeHandles/SafeProvOrNCryptKeyHandleUwp.cs

```diff
@@ -24,7 +37,14 @@ internal SafeProvOrNCryptKeyHandleUwp(IntPtr handle, bool ownsHandle, bool isNcr
 
         protected sealed override bool ReleaseHandle()
         {
-            if (_isNcrypt)
+            if (_parentHandle != null)
+            {
+                _parentHandle.DangerousRelease();
+                _parentHandle = null;
+                SetHandle(IntPtr.Zero);
```

Comment:
It shouldn't, since ReleaseHandle is only called after a interlocked decrement results in a 0 ref count.  (Unlike Dispose which could be called by multiple parallel callers)